pwiz_tools/Skyline/Model/DocSettings/SrmSettings.cs

```diff
@@ -2785,6 +2785,14 @@ public SrmSettingsDiff(SrmSettings settingsOld, SrmSettings settingsNew)
             // Force update if the bulk import has just completed
             else if (settingsOld.HasResults && settingsOld.MeasuredResults.IsResultsUpdateRequired)
                 DiffResults = true;
+
+            if (settingsNew.HasLibraries && !settingsNew.PeptideSettings.Libraries.IsLoaded &&
+                settingsNew.PeptideSettings.Libraries.Pick != PeptidePick.filter)
+            {
+                // If the libraries have not been loaded then we will not be able to determine which peptides
+                // should be chosen
+                DiffPeptides = false;

```

Comment:
I think you want to turn off all of DiffPeptides, DiffTransitionGroups, and DiffTransitions. Any of these could have disastrously bad results if applied before the libraries are loaded.